src/OnlineRetailPortal.Web/Validations/UpdateProductRequestValidator.cs

```diff
@@ -7,117 +7,117 @@
             .NotEmpty()
             .WithErrorCode(ErrorCode.MissingField())
             .WithMessage(Error.MissingField("Description"))
             .Length(4, 100)
             .WithErrorCode(ErrorCode.GreaterCharacter())
+            .When(x=>x.Description !=null)
             .WithMessage(Error.GreaterCharacter("Description", "4"));
 
             RuleFor(x => x.PurchasedDate)
             .LessThan(DateTime.Today)
             .WithErrorCode(ErrorCode.GreaterDate())
+            .When(x=>x.PurchasedDate == new DateTime())
             .WithMessage(Error.GreaterDate("PurchasedDate"));
 
-            When(x => x.PickupAddress != null && x.PickupAddress.Line1 != null &&
-            x.PickupAddress.City != null && x.PickupAddress.Pincode > 0 && x.PickupAddress.State != null,
+            When(x => x.PickupAddress != null && (x.PickupAddress.Line1 != null ||
+            x.PickupAddress.City != null || x.PickupAddress.Pincode >=0 || x.PickupAddress.State != null),
             () =>
             {
                 RuleFor(x => x.PickupAddress.Line1)
                 .Cascade(CascadeMode.StopOnFirstFailure)
-                .NotNull()
-                .WithErrorCode(ErrorCode.NullField())
-                .WithMessage(Error.NullField("Line1"))
-                .NotEmpty()
-                .WithErrorCode(ErrorCode.MissingField())
-                .WithMessage(Error.MissingField("Line1"))
                 .Length(2, 20)
                 .WithErrorCode(ErrorCode.GreaterCharacter())
                 .WithMessage(Error.GreaterCharacter("Line1", "2"));
 
-                RuleFor(x => x.PickupAddress.City)
+                RuleFor(x => x.PickupAddress.Pincode.ToString())
                 .Cascade(CascadeMode.StopOnFirstFailure)
-                .NotNull()
-                .WithErrorCode(ErrorCode.NullField())
-                .WithMessage(Error.NullField("City"))
-                .NotEmpty()
-                .WithErrorCode(ErrorCode.MissingField())
-                .WithMessage(Error.MissingField("City"));
+                .Must(x => x.Length == 6)
+                .WithErrorCode(ErrorCode.PincodeLength())
+                .When(x => x.PickupAddress.Pincode >= 0 );
 
-                RuleFor(x => x.PickupAddress.State)
-                .Cascade(CascadeMode.StopOnFirstFailure)
-                .NotNull()
-                .WithErrorCode(ErrorCode.NullField())
-                .WithMessage(Error.NullField("State"))
-                .NotEmpty()
-                .WithErrorCode(ErrorCode.MissingField())
-                .WithMessage(Error.MissingField("State"));
-
-                RuleFor(x => x.PickupAddress.Pincode)
-                .Cascade(CascadeMode.StopOnFirstFailure)
-                .NotEmpty()
-                .WithErrorCode(ErrorCode.MissingField())
-                .WithMessage(Error.MissingField("Pincode"));
             });
 
         }
+
+        private bool validateStatus(string status)
```

Comment:
again status is of type string and you had a class called Status, do Status class has any references?

Reply:
yes we are using Status as Enum in Core and Services